tron/serialize/runstate/dynamodb_state_store.py

```diff
@@ -47,38 +50,50 @@ def restore(self, keys) -> dict:
         vals = self._merge_items(first_items, remaining_items)
         return vals
 
-    def _get_items(self, keys: list) -> object:
-        items = []
+    def chunk_keys(self, keys: list) -> list:
+        """Generates a list of chunks of keys to be used to read from DynamoDB"""
+        # have a for loop here for all the key chunks we want to go over
+        cand_keys_chunks = []
         for i in range(0, len(keys), 100):
-            count = 0
-            cand_keys = keys[i : min(len(keys), i + 100)]
-            while True:
-                resp = self.client.batch_get_item(
-                    RequestItems={
-                        self.name: {
-                            "Keys": cand_keys,
-                            "ConsistentRead": True,
-                        },
-                    },
-                )
-                items.extend(resp["Responses"][self.name])
-                if resp["UnprocessedKeys"].get(self.name) and count < 10:
-                    cand_keys = resp["UnprocessedKeys"][self.name]["Keys"]
-                    count += 1
-                elif count >= 10:
-                    error = Exception(
-                        f"tron_dynamodb_restore_failure: failed to retrieve items with keys \n{cand_keys}\n from dynamodb\n{resp}"
+            # chunks of at most 100 keys will be in this list as there could be smaller chunks
+            cand_keys_chunks.append(keys[i : min(len(keys), i + 100)])
+        return cand_keys_chunks
+
+    def _get_items(self, table_keys: list) -> object:
+        items = []
+        # let's avoid potentially mutating our input :)
+        cand_keys_list = copy.copy(table_keys)
+        attempts_to_retrieve_keys = 0
+        while attempts_to_retrieve_keys < MAX_ATTEMPTS and len(cand_keys_list) != 0:
```

Comment:
fwiw: we're missing the old behavior of throwing an exception after we've exhausted retries

Reply:
I added it in the new commit